src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -23,6 +24,20 @@ public class PackageManagerExtension : IExtension, ILogSource, IExtensionSource
 
         public event Action<ILogMessage> MessageLogged;
 
+        private IWorkspaceModel currentWorkspace;
+
+        /// <summary>
+        /// Dictionary mapping a custom node functionID to the package that contains it.
+        /// Used for package dependency serialization.
+        /// </summary>
+        private Dictionary<Guid, PackageInfo> CustomNodePackageDictionary;
+
+        /// <summary>
+        /// Dictionary mapping the AssemblyName.FullName of an assembly to the package that contains it.
+        /// Used for package dependency serialization.
+        /// </summary>
+        private Dictionary<string, PackageInfo> NodePackageDictionary;
```

Comment:
why not use AssemblyName objects?

Reply:
`AssemblyName.Equals()` wasn't working as expected for dictionary keys here, so I just used `AssemblyName.FullName` which contains its Name and Version.